src/main/routes/course.js

```diff
@@ -6,145 +6,208 @@ var router = express.Router();
+                    {
+                        name: 'n/a',
+                        exceeds: 'n/a',
+                        meets: 'n/a',
+                        partially: 'n/a',
+                        not: 'n/a'
+                    },
+                    {
+                        name: 'n/a',
+                        exceeds: 'n/a',
+                        meets: 'n/a',
+                        partially: 'n/a',
+                        not: 'n/a'
+                    },
+                    {
+                        name: 'n/a',
+                        exceeds: 'n/a',
+                        meets: 'n/a',
+                        partially: 'n/a',
+                        not: 'n/a'
+                    },
+                    {
+                        name: 'n/a',
+                        exceeds: 'n/a',
+                        meets: 'n/a',
+                        partially: 'n/a',
+                        not: 'n/a'
+                    },
+                ],
+                artifacts: [
+                    {
+                        name: 'n/a',
+                        evaluations: [
+                            {
+                                index: 1,
+                                evaluation: [
+                                    {
+                                        metric: 1,
+                                        value: 6
+                                    },
+                                    {
+                                        metric: 2,
+                                        value: 6
+                                    },
+                                    {
+                                        metric: 3,
+                                        value: 6
+                                    },
+                                    {
+                                        metric: 4,
+                                        value: 6
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    };
+
+    res.render('base_template', {
+        title: 'CS498 Course Portfolio',
+        body: mustache.render('course/manage', course_info)
+    })
 }
 
+
+const setReadOnly = async (portfolio_id, readonly) => {
+    await Portfolio.query()
+        .where({ "id": portfolio_id })
+        .patch({ "readonly": readonly })
+}
+
+const sortPortfolios = (portfolios) => {
+    // very statically sorts the portfolios
+    const date = new Date();
+    const current_year = date.getFullYear();
+    const current_month = date.getMonth();
+
+    // very rudimentary semester divisions, only by month
+    let semester;
+    if (current_month < 5) {
+        semester = "SPRING";
+    } else if (current_month < 6) {
+        semester = "SUMMER 1"
+    } else if (current_month < 7) {
+        semester = "SUMMER 1"
+    } else if (current_month < 6) {
```

Comment:
Is this supposed to be current_month < 6? You have that same condition for a previous if statement setting the semester to "SUMMER 1".

Reply:
Good catch, fixed that